src/Sql/Vault/dbo/Stored Procedures/Cipher/Cipher_DeleteAttachment.sql

```diff
@@ -10,20 +10,59 @@ BEGIN
 
     DECLARE @UserId UNIQUEIDENTIFIER
     DECLARE @OrganizationId UNIQUEIDENTIFIER
+    DECLARE @CurrentAttachments NVARCHAR(MAX)
+    DECLARE @NewAttachments NVARCHAR(MAX)
 
+    -- Get current cipher data
     SELECT
         @UserId = [UserId],
-        @OrganizationId = [OrganizationId]
-    FROM 
+        @OrganizationId = [OrganizationId],
+        @CurrentAttachments = [Attachments]
+    FROM
         [dbo].[Cipher]
     WHERE [Id] = @Id
 
-    UPDATE
-        [dbo].[Cipher]
-    SET
-        [Attachments] = JSON_MODIFY([Attachments], @AttachmentIdPath, NULL)
-    WHERE
-        [Id] = @Id
+    -- If there are no attachments, nothing to do
+    IF @CurrentAttachments IS NULL
+    BEGIN
+        RETURN;
+    END
+
+    -- Validate the initial JSON
+    IF ISJSON(@CurrentAttachments) = 0
+    BEGIN
+        THROW 50000, 'Current initial attachments data is not valid JSON', 1;
+        RETURN;
+    END
+
+    -- Check if the attachment exists before trying to remove it
+    IF JSON_PATH_EXISTS(@CurrentAttachments, @AttachmentIdPath) = 0
```

Comment:
ℹ️ Switched to `JSON_PATH_EXISTS` as `JSON_VALUE` would return `NULL` for non-scalar results